example/bench/parameters.cpp

```diff
@@ -0,0 +1,111 @@
+#include <exception>
+#include <fstream>
+#include <string>
+
+#include <util/optional.hpp>
+
+#include <json/json.hpp>
+
+#include "parameters.hpp"
+
+double bench_params::expected_advance_time() const {
+    return cell.realtime_ratio * duration*1e-3 * num_cells;
+}
+unsigned bench_params::expected_spikes() const {
+    return num_cells * duration*1e-3 * cell.spike_freq_hz;
+}
+unsigned bench_params::expected_spikes_per_interval() const {
+    return num_cells * network.min_delay*1e-3/2 * cell.spike_freq_hz;
+}
+unsigned bench_params::expected_events() const {
+    return expected_spikes() * network.fan_in;
+}
+unsigned bench_params::expected_events_per_interval() const {
+    return expected_spikes_per_interval() * network.fan_in;
+}
+
+std::ostream& operator<<(std::ostream& o, const bench_params& p) {
+    o << "benchmark parameters:\n"
+      << "  name:          " << p.name << "\n"
+      << "  num cells:     " << p.num_cells << "\n"
+      << "  duration:      " << p.duration << " ms\n"
+      << "  fan in:        " << p.network.fan_in << " connections/cell\n"
+      << "  min delay:     " << p.network.min_delay << " ms\n"
+      << "  spike freq:    " << p.cell.spike_freq_hz << " Hz\n"
+      << "  cell overhead: " << p.cell.realtime_ratio << " ms to advance 1 ms\n";
+    o << "expected:\n"
+      << "  cell advance: " << p.expected_advance_time() << " s\n"
+      << "  spikes:       " << p.expected_spikes() << "\n"
+      << "  events:       " << p.expected_events() << "\n"
+      << "  spikes:       " << p.expected_spikes_per_interval() << " per interval\n"
+      << "  events:       " << p.expected_events_per_interval()/p.num_cells << " per cell per interval";
+    return o;
+}
+
+using arb::util::optional;
+using nlohmann::json;
+
+// Search a json object for an entry with a given name.
+// If found, return the value and remove from json object.
+template<typename T>
+optional<T> extract(const char* name, json& j) {
+    auto it = j.find(name);
+    if (it==j.end()) {
+        return arb::util::nullopt;
+    }
+    T value = std::move(*it);
+    j.erase(name);
+    return std::move(value);
+}
+
+bench_params read_options(int argc, char** argv) {
```

Comment:
Add comment about command line options, ie: 
`bench.exe <config-filename>?`
with config-filename referencing where we define the json variables

Reply:
Sure.